### DISCUSSION TITLE
Original LAN Modbus no longer supported?

### DISCUSSION THREAD

**Discussion by C00K50N** [nathanmarlor/foxess_em#245] [General]
I've just upgraded to v1.7.4 since our batteries got an unnecessary full charge Saturday night then failed to charge last night, I presumed due to the clocks changing to BST.  

But I'm not getting anywhere since the upgrade either - I got an error message relating to Modbus connection.  So I've tried deleting and reinstalling EM but it looks like it can no longer read data via HA-FoxESS-Modbus.  Is that right?  I've been using that via LAN up to now as I don't have a serial to LAN converter.

![image](https://user-images.githubusercontent.com/102669305/227885497-b154aac7-6ebb-411a-bb31-099e5ea842f3.png)

Is my only option now to buy a serial to LAN device?  I don't want to go back to the unreliable Fox Cloud connection.

Furthermore, in deleting / trying to reinstall EM I seem to have lost my Modbus connection.  I'm now updating HA Core and other bits to see if that solves the issues.

**Comment by nathanmarlor** [comment]
@C00K50N were you using direct LAN or through an RTU -> TCP box (i.e. W610)?

**Reply by C00K50N** [reply]
Direct from the LAN port on the inverter to my router.

**Reply by nathanmarlor** [reply]
That's odd, it should've never worked through Modbus as the charge time registers are different and the W610 uses holding registers vs. LAN using input registers.

**Reply by C00K50N** [reply]
Oh really?!  That's most odd, although I have to confess I don't really understand any of that technical stuff.  All I know is that it was working really well recently, to the point where I was happy not to upgrade anything.  But then the clocks changed and it all went to pot!

**Reply by nathanmarlor** [reply]
Yep, see -> https://github.com/nathanmarlor/foxess_em/discussions/224

**Reply by C00K50N** [reply]
Thanks, yes I was reading that one.  I figured I wouldn't need to do the same since mine was working, but I guess I'll need to head down that route too.  

Or... my HA runs on an old laptop, which I could place next to the inverter...  Is there any benefit of serial > USB rather than serial > LAN device?  Just thinking if I'm going to buy a device I may as well buy the right one one since I can place the HA hardware wherever.

**Reply by C00K50N** [reply]
Hang on, I think I've got confused @nathanmarlor... changing the times via your EM integration I've always done via the Cloud.  Sorry, I've had a mind melt!  It's just reading the sensors from the inverter I've been doing via the LAN Modbus integration, but changing times based on forecast was always Cloud.

Sorry for the confusion, I have reinstalled including my Cloud login details and it looks like we're back to normal.

**Reply by nathanmarlor** [reply]
Welcome to Monday :smiley:

**Reply by C00K50N** [reply]
Ha ha!  Yes, quite.  I obviously needed the whole morning to properly wake up!
